lib/cli/src/Cardano/CLI.hs

```diff
@@ -836,6 +839,27 @@ cmdWalletCreate = command "create" $ info (helper <*> cmd) $ mempty
                 (ApiT wName)
                 (ApiT wPwd)
 
+cmdWalletRestore
+    :: forall t. (DecodeAddress t, EncodeAddress t)
+    => Mod CommandFields (IO ())
+cmdWalletRestore = command "restore" $ info (helper <*> cmd) $ mempty
+    <> progDesc "Restore a wallet using a public account key."
+  where
+    cmd = fmap exec $ WalletCreateArgs
+        <$> portOption
+        <*> walletNameArgument
+        <*> poolGapOption
+    exec (WalletCreateArgs wPort wName wGap) = do
+        pubAccKey <- do
+            let prompt = "Please enter a 128 character public account key: "
+            let parser = fromText @AccountPublicKey
+            fst <$> getLine prompt parser
```

Comment:
What drives the choice for an interactive prompt here :thinking: ? 
It seems easier / more interoperable (i.e. piping) to have the command-line take the key as an argument here.

Reply:
Right, I suppose public key does not have to be "hidden" inside interactive prompt since it is public... (unlike mnemonic sentence for that matter)

Reply:
persuasive points: done in https://github.com/input-output-hk/cardano-wallet/pull/1421/commits/a2b4df64a33184d951009ff9c34e600da8d6fc3a